### PR TITLE
Refinement and elaboration of V types

### PR DESCRIPTION
**Pull request by jonsterling**
😎 

- [x] refinement rule for V type code
- [x] refinement rule for V type intro
- [x] refinement rule for V type elim
- [x] elaboration of V type code
- [x] elaboration of V intro
- [x] elaboration of V elim

### CODE REVIEW

**Comment by favonia** [comment]
OMG this looks much more Immortal now!

File: src/lib/Refiner.ml [hunk 1]
```diff
@@ -470,6 +470,33 @@ struct
...
+      let* vpcode = EM.lift_ev @@ Sem.eval pcode in
+      let* vcode = EM.lift_ev @@ Sem.eval code in
+      EM.lift_cmp @@
+      Sem.splice_tp @@
+      Splice.foreign_dim vr @@ fun r ->
+      Splice.foreign vpcode @@ fun pcode ->
+      Splice.foreign vcode @@ fun code ->
+      Splice.term @@
+      TB.pi (TB.tp_prf (TB.eq r TB.dim0)) @@ fun _ ->
+      TB.el @@ TB.Equiv.code_equiv (TB.ap pcode [TB.prf]) code
```
Comment (favonia) [inline src/lib/Refiner.ml:496]: Did we repeat the same code, say, 5 times already?

Comment (jonsterling) [reply src/lib/Refiner.ml:496]: Yes we have, but most of the "repeated" code is local paperwork (`Splice.foreign`), so I don't think it is a good candidate for factoring out.

Comment (favonia) [reply src/lib/Refiner.ml:496]: I wonder where we should put this... it's painful to see the same code everywhere LOL.

**Pull request merged**